Add TensorFlow.js

aaronhma opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

https://github.com/aaronhma/awesome-tensorflow-js

### About
This repository contains a curated list of dedicated resources to help you become a ninja at TensorFlow.js.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

aaronhma commented:
@sindresorhus I fixed everything and completed everything on the checklist. Thanks! :) 👍

uppusaikiran commented:
@aaronhma  Change title from **Add Awesome TensorFlow.js** to **Add TensorFlow.js** as per the guidelines.

aaronhma commented:
@uppusaikiran Thanks! I have already updated it.

aaronhma commented:
Reviewed PR: #1767 #1660

aaronhma commented:
@sindresorhus Can you please approve of this PR? Thanks! :)

sindresorhus commented:
The `About TensorFlow.js` section should be at the top, below the title, without the section header.

---

`Table of Contents` should not be in the `Contents` list.

---

I think the repo name should be `awesome-tensorflow-js` or `awesome-tensorflowjs`.

---

You cannot just copy https://github.com/aaronhma/awesome-tensorflow-tfjs/blob/master/CONTRIBUTING.md from Awesome. It needs to be adapted to your list. See other Awesome list for inspiration.

---

```
  README.md:87:3
  ✖   1:1   Main heading must be in title case                                                                         remark-lint:awesome-heading
  ✖   1:26  Invalid badge source                                                                                       remark-lint:awesome-badge
  ✖   9:1   ToC item "About TensorFlow.js" link "https://github.com/aaronhma/awesome-tensorflow-tfjs#about" not found  remark-lint:awesome-toc
```

https://github.com/sindresorhus/awesome-lint

aaronhma commented:
> The `About TensorFlow.js` section should be at the top, below the title, without the section header.

Done. Removed section header, and moved about section below the title.

> `Table of Contents` should not be in the `Contents` list.

Done. Created a new and updated Table of Contents separated by categories.

> I think the repo name should be `awesome-tensorflow-js` or `awesome-tensorflowjs`.

Done. Rename the repo to [awesome-tensorflow-js](https://github.com/aaronhma/awesome-tensorflow-js)

> You cannot just copy https://github.com/aaronhma/awesome-tensorflow-tfjs/blob/master/CONTRIBUTING.md from Awesome. It needs to be adapted to your list. See other Awesome list for inspiration.``

Done. The [contribution guide](https://github.com/aaronhma/awesome-tensorflow-js/blob/master/CONTRIBUTING.md) is fully updated with detailed descriptive instruction. 

> https://github.com/sindresorhus/awesome-lint

Done. Lint free used *awesome-lint* to lint the list. 

![Screen Shot 2020-06-02 at 11 08 35 AM](https://user-images.githubusercontent.com/25288451/83554316-6f3af680-a4c1-11ea-9645-d770eadb0b00.png)

@sindresorhus Please, re-review! Thank you very much for your detailed instruction and it helps me a lot as a first-timer contributor!

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1268079326248886277
